crates/astria-sequencer/src/grpc/storage/values/rollup_ids.rs

```diff
@@ -0,0 +1,82 @@
+use std::{
+    borrow::Cow,
+    fmt::{
+        self,
+        Display,
+        Formatter,
+    },
+};
+
+use astria_core::primitive::v1::RollupId as DomainRollupId;
+use astria_eyre::eyre::bail;
+use borsh::{
+    BorshDeserialize,
+    BorshSerialize,
+};
+use itertools::Itertools as _;
+use telemetry::display::base64;
+
+use super::{
+    Value,
+    ValueImpl,
+};
+
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub(super) struct RollupId<'a>(Cow<'a, [u8; 32]>);
+
+impl<'a> Display for RollupId<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        base64(self.0.as_slice()).fmt(f)
+    }
+}
+
+impl<'a> From<&'a DomainRollupId> for RollupId<'a> {
+    fn from(rollup_id: &'a DomainRollupId) -> Self {
+        RollupId(Cow::Borrowed(rollup_id.as_bytes()))
+    }
+}
+
+impl<'a> From<RollupId<'a>> for DomainRollupId {
+    fn from(rollup_id: RollupId<'a>) -> Self {
+        DomainRollupId::new(rollup_id.0.into_owned())
+    }
+}
+
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub(in crate::grpc) struct RollupIds<'a>(Vec<RollupId<'a>>);
+
+impl<'a> Display for RollupIds<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "[{}]", self.0.iter().join(", "))
```

Comment:
Can we avoid the allocation here and just:
```suggestion
        f.write_str("[")?;
        let mut ids = self.0.iter();
        if let Some(id) = ids.next();
            id.fmt(f)?;
        }
        for id in ids {
            f.write_str(", ")?;
            id.fmt(f)?;
        }
        f.write_str("]")?;
        Ok(())
    }
```
    
            
        write!(f, "[{}]", self.0.iter().join(", "))

Reply:
This code is now deleted (using Debug instead).